server/src/main/java/com/paypal/selion/proxy/SeLionSauceProxy.java

```diff
@@ -26,33 +22,27 @@
 import java.util.concurrent.locks.ReentrantLock;
 import java.util.logging.Level;
 
+import com.paypal.selion.utils.SauceLabsRestApiFactory;
 import org.apache.commons.lang.StringUtils;
-import org.apache.http.HttpStatus;
 import org.openqa.grid.common.RegistrationRequest;
 import org.openqa.grid.internal.Registry;
 import org.openqa.grid.internal.TestSession;
 import org.openqa.grid.selenium.proxy.DefaultRemoteProxy;
 
-import com.google.gson.JsonObject;
-import com.google.gson.JsonParser;
-import com.google.gson.JsonSyntaxException;
 import com.paypal.selion.logging.SeLionGridLogger;
-import com.paypal.selion.utils.SauceConfigReader;
 
 /**
  * This customized version of {@link DefaultRemoteProxy} basically helps redirect all traffic to the SauceLabs cloud.
- * 
  */
 public class SeLionSauceProxy extends DefaultRemoteProxy {
 
     private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(SeLionSauceProxy.class);
-
-    private int maxTestCase = -1;
-
+    private final int maxConcurrent;
     private final Lock accessLock = new ReentrantLock();
 
     public SeLionSauceProxy(RegistrationRequest request, Registry registry) {
         super(request, registry);
+        maxConcurrent = getMaxNumberOfConcurrentTestSessions();
```

Comment:
Yes. This is intentional. We'll document it. The account configuration shouldn't change often in this area. I'll look into adding a "restart sauce node" control in the SeLion Grid Management Console